src/main/java/com/group6/harmoniq/controllers/SpotifyController.java

```diff
@@ -0,0 +1,157 @@
+
+// Copyright 2012-2016 Spotify AB
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//Original Source: https://github.com/spotify/web-api-examples
+//This file was modified from the original source by Christian Chidiac on June 30, 2024.
+//This code was converted from JavaScript to Java by Christian Chidiac on June 30, 2024.
+
+package com.group6.harmoniq.controllers;
+
+import java.security.SecureRandom;
+import java.nio.charset.StandardCharsets;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.http.*;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.CookieValue;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+import org.springframework.ui.Model;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletResponse;
+
+
+@Controller
+public class SpotifyController {
+    
+    String client_id = System.getenv("SPOTIFY_CLIENT_ID");
+    String client_secret = System.getenv("SPOTIFY_CLIENT_SECRET");
+    String redirect_uri = "https://harmoniq-1.onrender.com/callback";
+    final String stateKey = "spotify_auth_state";
+    
+    
+    //redirects to Spotify API's Authorization endpoint
+   @GetMapping("/login")
+   public String login(HttpServletResponse response) {
+    SecureRandom secureRandom;
+    try {
+        secureRandom = SecureRandom.getInstanceStrong();
+    } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+
+        String state = generateRandomString(secureRandom, 16);
+        Cookie stateCookie = new Cookie(stateKey, state);
+        response.addCookie(stateCookie);
+        String scope = "user-read-private user-read-email user-follow-read";
+
+        //Requesting Authorization
+        String authUrl = UriComponentsBuilder.fromHttpUrl("https://accounts.spotify.com/authorize")
```

Comment:
Lets add the main URL as a property of this controller so that we only need to add the path here and on line 93